Define _XOPEN_SOURCE=500 to permit headers to prototype strdup() with -std=99

../mesa/program/symbol_table.c: In function ‘_mesa_symbol_table_add_symbol’:
../mesa/program/symbol_table.c:341:8: warning: implicit declaration of function ‘strdup’ [-Wimplicit-function-declaration]
        hdr->name = strdup(name);
        ^
../mesa/program/symbol_table.c:341:18: warning: assignment makes pointer from integer without a cast [enabled by default]
        hdr->name = strdup(name);
                  ^
../mesa/program/symbol_table.c: In function ‘_mesa_symbol_table_add_global_symbol’:
../mesa/program/symbol_table.c:398:19: warning: assignment makes pointer from integer without a cast [enabled by default]
         hdr->name = strdup(name);

If we fail to do this, the implicit int return of strdup() being converted to a char * causes builtin_compiler to fail to
function correctly on 64-bit cygwin...

--- a/configure.ac
+++ b/configure.ac
@@ -159,20 +159,13 @@ solaris*)
     DEFINES="$DEFINES -DHAVE_PTHREAD -DSVR4"
     ;;
 cygwin*)
-    DEFINES="$DEFINES -DHAVE_PTHREAD"
+    DEFINES="$DEFINES -D_XOPEN_SOURCE=500 -DHAVE_PTHREAD"
     ;;
 esac

 dnl Add flags for gcc and g++
 if test "x$GCC" = xyes; then
-    case "$host_os" in
-    cygwin*)
-        CFLAGS="$CFLAGS -Wall -std=gnu99"
-        ;;
-    *)
-        CFLAGS="$CFLAGS -Wall -std=c99"
-        ;;
-    esac
+    CFLAGS="$CFLAGS -Wall -std=c99"

     # Enable -Werror=implicit-function-declaration and
     # -Werror=missing-prototypes, if available, or otherwise, just
